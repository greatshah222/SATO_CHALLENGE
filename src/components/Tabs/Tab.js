import React from 'react';
import classNames from 'classnames';
import { makeStyles } from '@material-ui/core/styles';
import MuiTab from '@material-ui/core/Tab';
import * as tokens from '../../.generated/styles';

const useStyles = makeStyles(theme => ({
  root: {
    ...theme.typography.button,
    color: tokens.textGrey,
    minWidth: tokens.spacingGargantual,
    paddingTop: tokens.spacingLarge,
    paddingBottom: tokens.spacingLarge,
    whiteSpace: 'nowrap',
  },
  selected: {
    color: tokens.colorBlack,
  },
  labelContainerLarge: {
    paddingTop: tokens.spacing2Xl,
    paddingBottom: tokens.spacing2Xl,
    [theme.breakpoints.down('sm')]: {
      paddingTop: tokens.spacingLarge,
      paddingBottom: tokens.spacingLarge,
    },
  },
  extraRightPadding: {
    paddingRight: tokens.spacingXl,
  },
  icon: {
    right: 0,
    top: '50%',
    transform: 'translateY(-50%)',
  },
  labelIcon: {
    paddingTop: 0,
    paddingBottom: 0,
  },
}));

const Tab = ({ key, label, icon, largePadding, numOfNewMessages, ...props }) => {
  const classes = useStyles();
  return (
    <MuiTab
      key={key}
      label={label}
      icon={icon}
      classes={{
        root: classNames(classes.root, {
          [classes.labelContainerLarge]: largePadding,
          [classes.extraRightPadding]: numOfNewMessages && numOfNewMessages > 0,
        }),
        selected: classes.selected,
        labelIcon: classes.labelIcon,
      }}
      disableRipple
      {...props}
    />
  );
};

export default Tab;
