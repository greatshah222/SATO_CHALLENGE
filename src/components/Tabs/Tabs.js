import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import MuiTabs from '@material-ui/core/Tabs';
import NewMessageDot from '../../icons/NewMessageDot';

import Tab from './Tab';

const styles = theme => ({
  icon: {
    right: 0,
    top: '50%',
    transform: 'translateY(-50%)',
  },
});

const Tabs = ({ classes, tabs, handleChange, defaultValue, largePadding, children }) => (
  <MuiTabs
    value={defaultValue}
    variant="scrollable"
    indicatorColor="primary"
    onChange={handleChange}
    scrollButtons="off"
  >
    {!children
      ? tabs &&
        tabs.map(tab => (
          <Tab
            key={tab.key}
            label={tab.label}
            largePadding={largePadding}
            numOfNewMessages={tab.numOfNewMessages}
            icon={
              tab.numOfNewMessages && tab.numOfNewMessages > 0 ? (
                <NewMessageDot className={classes.icon}>{tab.numOfNewMessages}</NewMessageDot>
              ) : null
            }
            disableRipple
          />
        ))
      : children}
  </MuiTabs>
);

Tabs.propTypes = {
  handleChange: PropTypes.func.isRequired,
  defaultValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  classes: PropTypes.object,
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
      label: PropTypes.string.isRequired,
    })
  ),
  largePadding: PropTypes.bool,
  children: PropTypes.node,
};

export default withStyles(styles)(Tabs);
