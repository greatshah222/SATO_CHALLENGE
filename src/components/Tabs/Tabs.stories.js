import React from 'react';
import PropTypes from 'prop-types';
import { storiesOf } from '@storybook/react';
import { muiTheme } from 'storybook-addon-material-ui';

import Tabs, { Tab } from './index';
import theme from '../../theme';

const tabsContent = [
  { key: 0, label: 'Helsinki' },
  { key: 1, label: 'Turku' },
  { key: 2, label: 'Tampere' },
  { key: 3, label: '1h' },
  { key: 4, label: '2h' },
  { key: 5, label: '3h' },
];

const tabsContentWithMessages = [
  { key: 0, label: 'Helsinki', numOfNewMessages: 2 },
  { key: 1, label: 'Turku' },
  { key: 2, label: 'Tampere', numOfNewMessages: 1 },
];

const ExampleTabs = ({ largePadding, newMessages }) => {
  const [index, setIndex] = React.useState(0);

  const onHandleChange = (e, index) => {
    setIndex(index);
  };

  if (newMessages) {
    return (
      <Tabs
        defaultValue={index}
        tabs={tabsContentWithMessages}
        handleChange={onHandleChange}
        largePadding={largePadding}
      />
    );
  } else {
    return <Tabs defaultValue={index} tabs={tabsContent} handleChange={onHandleChange} largePadding={largePadding} />;
  }
};

ExampleTabs.propTypes = {
  largePadding: PropTypes.bool,
};

const TabWithChildren = ({ defaultValue, children, largePadding }) => {
  const [index, setIndex] = React.useState(defaultValue);

  const handleChange = (e, idx) => {
    setIndex(idx);
  };

  return (
    <Tabs defaultValue={index} handleChange={handleChange} largePadding={largePadding}>
      {children}
    </Tabs>
  );
};

TabWithChildren.propTypes = {
  largePadding: PropTypes.bool,
  defaultValue: PropTypes.number,
  children: PropTypes.node,
};

storiesOf('Tabs', module)
  .addDecorator(muiTheme(Object.values(theme)))
  .addWithJSX('default', () => <ExampleTabs />)
  .addWithJSX('large padding', () => <ExampleTabs largePadding />)
  .addWithJSX('With new message icon', () => <ExampleTabs newMessages />)
  .addWithJSX('With tabs as children', () => (
    <TabWithChildren defaultValue={0}>
      <Tab label={'Tab1'} largePadding={true} />
      <Tab label={'Tab2'} largePadding={true} />
      <Tab label={'Tab3'} largePadding={true} />
    </TabWithChildren>
  ));
