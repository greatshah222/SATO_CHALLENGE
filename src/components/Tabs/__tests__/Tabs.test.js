import React from 'react';
import { shallow } from 'enzyme';
import Tabs from '../Tabs';

describe('Tabs', () => {
  it('renders correctly', () => {
    const tree = shallow(<Tabs />);
    expect(tree).toMatchSnapshot();
  });
});
