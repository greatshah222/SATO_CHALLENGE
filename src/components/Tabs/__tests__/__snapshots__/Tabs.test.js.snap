// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly 1`] = `
<Tabs
  classes={
    Object {
      "icon": "Tabs-icon-1",
    }
  }
/>
`;
